config/confighttp/confighttp.go

```diff
@@ -105,6 +105,9 @@ type ClientConfig struct {
 	HTTP2PingTimeout time.Duration `mapstructure:"http2_ping_timeout"`
 	// Cookies configures the cookie management of the HTTP client.
 	Cookies *CookiesConfig `mapstructure:"cookies"`
+
+	// Maximum number of redirections to follow, if not defined, the Client uses its default policy, which is to stop after 10 consecutive requests.
+	MaxRedirects *int `mapstructure:"max_redirects"`
```

Comment:
I think we can even keep it `int` with `10` by default. That should be cleaner. The only problem is it'll be a significant breaking change (no redirects) for a component that doesn't use `NewDefaultClientConfig`, but I don't think we have any of those in contrib. Maybe we can check... All components should use that anyway. WDYT?

Reply:
I am totally open to adding this, but I am concert about receivers initiating the config without using the default method (max_redirects will be set to 0). Also, the only difference is that maybe we should align with the Go library error message after 10 redirects:

```go
func defaultCheckRedirect(req *Request, via []*Request) error {
	if len(via) >= 10 {
		return errors.New("stopped after 10 redirects")
	}
	return nil
}
```

There is this ongoing effort to use the `NewDefaultClientConfig`: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35457

Reply:
Different message isn't a problem I thought. I would like it to be merged without a pointer after all exporters use the default constructor. This is the pattern we use in other configs I believe.

Reply:
Sounds reasonable, let's keep it pointer free :) https://github.com/open-telemetry/opentelemetry-collector/pull/10877/commits/e6b62bb678915f56f5b8c9ee1512c3d08757808e

Should we wait until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35457 is done?

Reply:
contrib tests failing due https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36681